### ISSUE TITLE
Strange error

### ISSUE THREAD

**Issue by walseb** [minad/jinx#93]
Hello!

I always have `debug-on-error` enabled. I did something, and up popped the following error relating to jinx.

```
Debugger entered--Lisp error: (wrong-type-argument unicode-string-p "\211PNG")
  jinx--mod-check(#<user-ptr ptr=0x17d38f00 finalizer=0x7f3c574d6340> "\211PNG")
  jinx--word-valid-p(1)
  jinx--check-region(1 1499 nil)
  jinx--check-pending(1 1499)
  jinx--timer-handler()
  apply(jinx--timer-handler nil)
  timer-event-handler([t 0 0 200000 nil jinx--timer-handler nil idle 0 nil])
```

It seems to be related to some strange unicode character `\211PNG`. I did have buffers open with raw PNG files, as well as dired buffers with PNG files in them. I might have Jinx enabled in the PNG buffer, and so perhaps it tried to check the raw byte data? I'm running the latest version of Jinx available on git.

Thoughts?

**Comment by minad** [comment]
We could catch this error. On the other hand, enabling jinx in non-text buffers is not the right thing to do. global-jinx-mode should not enable Jinx in such buffers. Always enabling debug on error is not a good idea btw.

**Comment by walseb** [comment]
Thanks for the reply! You're right, probably that error shouldn't be caught since that character can probably only appear in raw encoding, not in UTF8.

> Always enabling debug on error is not a good idea btw, since it can change the behavior of code which uses condition-case-unless-debug.

That's interesting, I didn't know that. It's annoying to end up in a recursive edit so often as well. Do you know if there's a way to log errors into the warnings buffer instead of debugging them?

**Comment by daanturo** [comment]
> Hi! We could catch this error caused by non-Unicode character codes. On the other hand, enabling jinx in non-text buffers is not the right thing to do. Therefore this is not a bug. global-jinx-mode should not enable Jinx in such buffers.

This problem also happens when visiting `savehist-file` with `global-jinx-mode` enabled.

In that file, some variables are saved with string data who contains troublesome characters when saved such as `consult--buffer-history`.

I attached a minimal example trimmed down from `savehist-file` (the ".txt" extension is appended so that it's uploadable). It's not the exactly the same as `savehist-file` as I selected `no-conversion` when saving, but still triggers the timer's error.

Maybe `global-jinx-mode` should exclude such files and/or codings?


[jinx-error.txt](https://github.com/minad/jinx/files/12736265/jinx-error.txt)
